src/Microsoft.Extensions.Configuration.FileExtensions/FileConfigurationExtensions.cs

```diff
@@ -26,38 +93,33 @@ public static IConfigurationBuilder SetBasePath(this IConfigurationBuilder confi
                 throw new ArgumentNullException(nameof(basePath));
             }
 
-            configurationBuilder.Properties["BasePath"] = basePath;
-
-            return configurationBuilder;
+            return builder.SetFileProvider(new PhysicalFileProvider(basePath));
         }
 
         /// <summary>
-        /// Gets the base path to discover files in for file-based providers.
+        /// Adds a file configuration source to <paramref name="builder"/>.
         /// </summary>
-        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
-        /// <returns>The base path.</returns>
-        public static string GetBasePath(this IConfigurationBuilder configurationBuilder)
+        /// <typeparam name="TSource">The type of the file source.</typeparam>
+        /// <param name="builder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="configureSource">Configures the <see cref="IConfigurationSource"/> to add.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddFileSource<TSource>(this IConfigurationBuilder builder, Action<TSource> configureSource)
```

Comment:
Remove this - it's just another way of doing what the extension methods do, but it's less discoverable because you need to know the TSource in advance.